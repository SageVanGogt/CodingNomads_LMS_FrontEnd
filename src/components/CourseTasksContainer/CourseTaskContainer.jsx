import React, { Component } from 'react';
import { render } from 'react-dom';
import { SortableContainer, SortableElement, arrayMove } from 'react-sortable-hoc';
import { CourseTaskCard } from '../CourseTaskCard/CourseTaskCard';


const SortableItem = SortableElement(({ task }) =>
  <CourseTaskCard {...task} />
);

const SortableList = SortableContainer(({ tasks }) => {
  return (
    <ul>
      {tasks.map((task, index) => (
        <SortableItem key={`item-${index}`} index={index} task={task} />
      ))}
    </ul>
  );
});

export class CourseTaskContainer extends Component {
  constructor(props) {
    super(props)

    this.state = {
      tasks: []
    };
  }

  shouldCancelStart = (e) => {
    // Cancel sorting if the event target is an `input`, `textarea`, `select` or `option`
    if (['input', 'textarea', 'select', 'option', 'img', 'image'].indexOf(e.target.tagName.toLowerCase()) !== -1) {
      return true; // Return true to cancel sorting
    }
  }

  onSortEnd = ({ oldIndex, newIndex }) => {
    let tasks = arrayMove(this.props.tasks, oldIndex, newIndex);
    this.props.rearrangeTasks(tasks);
  };

  render() { 
    return (
      <div className='tasksArea'>
        <SortableList tasks={this.props.tasks} onSortEnd={this.onSortEnd} shouldCancelStart={this.shouldCancelStart} />
      </div>
    );
  }
}