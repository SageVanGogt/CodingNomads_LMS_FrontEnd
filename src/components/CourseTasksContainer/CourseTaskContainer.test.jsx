import React from 'react';
import { shallow } from 'enzyme';
import {CourseTaskContainer} from './CourseTaskContainer';

describe('CourseTaskContainer', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<CourseTaskContainer />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});