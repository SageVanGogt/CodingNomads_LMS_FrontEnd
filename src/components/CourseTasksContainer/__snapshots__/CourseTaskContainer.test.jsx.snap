// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseTaskContainer should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CourseTaskContainer />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <sortableList
        axis="y"
        deleteTask={undefined}
        distance={0}
        getHelperDimensions={[Function]}
        hideSortableGhost={true}
        lockOffset="50%"
        lockToContainerEdges={false}
        onSortEnd={[Function]}
        pressDelay={0}
        pressThreshold={5}
        shouldCancelStart={[Function]}
        tasks={undefined}
        transitionDuration={300}
        useWindowAsScrollContainer={false}
/>,
      "className": "tasksArea",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "axis": "y",
        "deleteTask": undefined,
        "distance": 0,
        "getHelperDimensions": [Function],
        "hideSortableGhost": true,
        "lockOffset": "50%",
        "lockToContainerEdges": false,
        "onSortEnd": [Function],
        "pressDelay": 0,
        "pressThreshold": 5,
        "shouldCancelStart": [Function],
        "tasks": undefined,
        "transitionDuration": 300,
        "useWindowAsScrollContainer": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <sortableList
          axis="y"
          deleteTask={undefined}
          distance={0}
          getHelperDimensions={[Function]}
          hideSortableGhost={true}
          lockOffset="50%"
          lockToContainerEdges={false}
          onSortEnd={[Function]}
          pressDelay={0}
          pressThreshold={5}
          shouldCancelStart={[Function]}
          tasks={undefined}
          transitionDuration={300}
          useWindowAsScrollContainer={false}
/>,
        "className": "tasksArea",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "axis": "y",
          "deleteTask": undefined,
          "distance": 0,
          "getHelperDimensions": [Function],
          "hideSortableGhost": true,
          "lockOffset": "50%",
          "lockToContainerEdges": false,
          "onSortEnd": [Function],
          "pressDelay": 0,
          "pressThreshold": 5,
          "shouldCancelStart": [Function],
          "tasks": undefined,
          "transitionDuration": 300,
          "useWindowAsScrollContainer": false,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
